tests/MongoDB.Driver.Examples/ChangeStreamExamples.cs

```diff
@@ -107,22 +108,22 @@ public void ChangeStreamExample3()
                 })
                 .Start();
 
-                var enumerator = inventory.Watch().ToEnumerable().GetEnumerator();
+                previousCursor = inventory.Watch(new ChangeStreamOptions { BatchSize = 2 });
+                var enumerator = previousCursor.ToEnumerable().GetEnumerator();
                 enumerator.MoveNext();
-                lastChangeStreamDocument = enumerator.Current;
             }
 
             {
                 // Start Changestream Example 3
-                var resumeToken = lastChangeStreamDocument.ResumeToken;
+                var resumeToken = previousCursor.GetResumeToken();
                 var options = new ChangeStreamOptions { ResumeAfter = resumeToken };
                 var enumerator = inventory.Watch(options).ToEnumerable().GetEnumerator();
                 enumerator.MoveNext();
                 var next = enumerator.Current;
                 enumerator.Dispose();
                 // End Changestream Example 3
 
-                next.FullDocument.Should().Be(documents[1]);
+                next.FullDocument.Should().Be(documents[2]);
```

Comment:
The fact that this changed from 1 to 2 means there is a bug in the sample code.

This example code is terrible... but that's not your fault of course. It was terrible when it was first written. No one should be consuming a change stream this way.

But I think the changes you have made have introduced a bug.

By mixing the use of `IEnumerator` and `IAsyncCursor` you have ended up with the wrong resume token.

I can't really tell from the JIRA ticket what you are supposed to change. How did you determine what needed to be done?

Reply:
I agree that the fact that the index changed from 1 to 2 indicates that there is a bug.
I restored the variable `lastChangeStrangeDocument` and stepped through this example with a debugger:
`lastChangeStreamDocument.ResumeToken` and `previousCursor.GetResumeToken()` have different values: I'm unsure, but my guess is that it's because the cursor technically fetched two documents and thus should resume from the third, where as the lastChangeStreamDocument (as written in the example) held the document with `{x:1} ` @rstam  Does this explanation sound feasible to you? 

I'm not sure if there's a way to use the cursor's resume token such that the next document that enumerator returns  is `{x:2}`  because our batchsize is  2. If we can't do this, perhaps the example should call `MoveNext` twice as to completely consume the first batch?

Reply:
The root problem is that the example code is mixing `IAsyncCursor` and `IEnumerator`.

The enumerator is out of sync with the cursor. The enumerator moves one document at a time and the cursor moves one batch at a time.

I recall having a discussion about the interplay between the enumerator and the cursor when `GetResumeToken` was implemented for the C# driver and we decided that we would not support resuming when using `IEnumerator`, only when using `IAsyncCursor`.

So the example needs to be changed to not use `IEnumerator` at all.

Reply:
I mentioned when this example was first written that it was not a good example. But we were given some sample code and told to translate verbatim to C# and that's what resulted.

The smallest possible change to this example to get it to work with `GetResumeToken` is to eliminate all use of the enumerator and work directly with the cursor.

It could look like this:

    var documents = new[]
    {
        new BsonDocument("x", 1),
        new BsonDocument("x", 2)
    };

    IChangeStreamCursor<ChangeStreamDocument<BsonDocument>> previousCursor;
    {
        new Thread(() =>
        {
            Thread.Sleep(TimeSpan.FromMilliseconds(100));
            inventory.InsertMany(documents);
        })
        .Start();

        previousCursor = inventory.Watch(new ChangeStreamOptions { BatchSize = 1 });
        while (previousCursor.MoveNext() && previousCursor.Current.Count() == 0) { } // keep calling MoveNext until we've read the first batch
    }

    {
        // Start Changestream Example 3
        var resumeToken = previousCursor.GetResumeToken();
        var options = new ChangeStreamOptions { ResumeAfter = resumeToken };
        var cursor = inventory.Watch(options);
        cursor.MoveNext();
        var next = cursor.Current.First();
        cursor.Dispose();
        // End Changestream Example 3

        next.FullDocument.Should().Be(documents[1]);
    }
}

Reply:
Done